src/popup/components/TransactionOverview.vue

```diff
@@ -5,106 +5,131 @@
     :sender="transaction.sender"
     :recipient="transaction.recipient"
     :tx-function="transaction.function"
+    :tx="tx"
   />
 </template>
 
-<script>
-import { mapState, mapGetters } from 'vuex';
+<script lang="ts">
+import {
+  computed,
+  defineComponent,
+  ref,
+  onMounted,
+} from '@vue/composition-api';
 import { SCHEMA } from '@aeternity/aepp-sdk';
-import { watchUntilTruthy } from '../utils';
+import { useGetter, useState } from '../../composables/vuex';
+import {
+  watchUntilTruthy,
+  TX_FUNCTION_DEPOSIT,
+  TX_FUNCTION_SENT,
+  TX_FUNCTION_RECEIVED,
+} from '../utils';
+import { IAccount } from '../../types';
+
 import TransactionInfo from './TransactionInfo.vue';
 
-export default {
+export default defineComponent({
+  name: 'TransactionOverview',
   components: { TransactionInfo },
   props: {
     tx: { type: Object, required: true },
     isDex: { type: Boolean, default: false },
   },
-  data: () => ({ name: '' }),
-  computed: {
-    ...mapGetters(['getTxType', 'getTxDirection', 'getExplorerPath', 'getDexContracts']),
-    ...mapGetters('names', ['getPreferred']),
-    ...mapState({
-      account(_, { account }) {
-        return {
-          ...account,
-          label: this.$t('transaction.overview.accountAddress'),
-          url: this.getExplorerPath(account.address),
-        };
-      },
-      isDexRecipient() {
-        return [...this.getDexContracts.router, ...this.getDexContracts.wae]
-          .includes(this.tx?.contractId);
-      },
-    }),
-    transaction() {
-      switch (this.txType) {
+  setup(props, { root }) {
+    const name = ref<string>();
+    const middleware = useState('middleware');
+    const getTxType = useGetter('getTxType');
+    const getTxDirection = useGetter('getTxDirection');
+    const getExplorerPath = useGetter('getExplorerPath');
+    const getDexContracts = useGetter('getDexContracts');
+    const getPreferredName = useGetter('names/getPreferred');
+    const account = useGetter<IAccount>('account');
+    const parsedAccount = computed<IAccount>(() => ({
```

Comment:
For the computed properties I suggest to use the type for the return. So in this case:
```
computed((): IAccount => ...
```
But in this case you won't be able to use the `IAccount` because of missing `label` and `url`. So new (eg.: IAccountExtended`) type should be created.